File: backend/routes/barber.js

/*jshint esversion: 6 */
const Barber = require('../database/models/barber');

var express = require('express');
var barberRouter = express.Router();
const multer = require('multer');

const storage = multer.diskStorage({
    destination: function(req, file, cb) {
        cb(null, './uploads/avatar/');
    },
    filename: function(req, file, cb) {
        cb(null, new Date().toISOString().replace(/:/g, '-') + '_' + file.originalname);
    }
});

const fileFilter = (req, file, cb) => {
    // reject a file
    if ((file.mimetype === 'image/jpeg') || (file.mimetype === 'image/png') || (file.mimetype === 'image/jpg')) {
        cb(null, true);
    } else {
        cb(new Error('File extention is not supported ' + file.mimetype), false);
    }
};
const upload = multer({
    storage: storage, 
    limits: {
        fileSize: 1024*1024*5
    },
    fileFilter: fileFilter
});

// Barber
barberRouter.get('/:barberId', (req, res) => {
    Barber.find({_id:req.params.barberId})
        .then(barbers => res.send(barbers))
        .catch((error) => console.log(error));
});

barberRouter.get('/', (req, res) => {
    Barber.find({})
        .then(barbers => res.send(barbers))
        .catch((error) => console.log(error));
});

barberRouter.post('/', upload.single('avatar'), (req, res) => {
    var strAvatarPath = "";
    if(req.file){
        console.log(req.file);
        strAvatarPath = req.file.path;
    }
    
    console.log(req.body);
    //var strBody = JSON.parse(JSON.stringify(req.body));
    
    const barber = new Barber({
        _salonId: req.params.salonId,
        username: req.body.username,
        firstname: req.body.firstname,
        lastname: req.body.lastname,
        phone: req.body.phone,
        email: req.body.email,
        dob: req.body.dob,
        gender: req.body.gender,
        profile: req.body.profile,
        avatar: strAvatarPath,

    });
    
    barber.save()
    .then(savedBarber => res.send(savedBarber))
    .catch((error) => console.log(error));
});


barberRouter.delete('/:barberId', (req, res) => {
    Barber.findByIdAndDelete({_id:req.params.barberId})
        .then(barber => res.send(barber))
        .catch((error) => console.log(error));
});

module.exports = barberRouter;

File: mobile/salonmobile/lib/models/Barber.dart
import 'package:flutter/material.dart';

class Barber {
  final String id;
  final String userId;
  final String salonId;
  final String salonName;
  final String userName;
  final String firstname, lastname;
  final String phone;
  final String email;
  final String dob;
  final String gender;
  final String profile;
  final String avatar;
  final String hometown;
  final String idcard;
  final double rating;

  Barber(
      {@required this.id,
      @required this.salonId,
        this.userId = "",
        this.salonName="",
        this.userName="",
        this.firstname="", this.lastname="",
        this.phone="",
        this.email="",
        this.dob="",
        this.gender="",
        this.profile="",
        this.avatar="",
        this.hometown="",
        this.idcard = "",
        this.rating = 0});

  factory Barber.fromJson(Map<String, dynamic> json) {
    double rating = 0;
    String avatar="assets/images/default-avatar.png", firstN="", lastN="", profile="";
    if (json['rating'] != null)
      rating = double.parse(json['rating']);
    if(json['avatar'] != null){
      avatar = json['avatar'];
    }
    if(json['firstname'] != null){
      firstN = json['firstname'];
    }
    if(json['lastname'] != null){
      lastN = json['lastname'];
    }
    if(json['profile'] != null){
      profile = json['profile'];
    }

    return Barber(
        id: json['id'],
        salonId: json['salonId'],
        firstname: firstN,
        lastname: lastN,
        profile: profile,
        rating: rating,
        avatar: avatar);
  }
}
